commit 3fb24230a2369513b1f36cba3d772bc517265e81
Author: Benda Xu <heroxbd@gentoo.org>
Date:   Wed Dec 10 10:45:31 2014 +0900

    In Prefix, let env-update call ldconfig if it exists.
    
    It does not affect prefix-rpath, and handles prefix-libc(RAP) correctly.

diff --git a/pym/portage/util/env_update.py b/pym/portage/util/env_update.py
index ace492c..4437881 100644
--- a/pym/portage/util/env_update.py
+++ b/pym/portage/util/env_update.py
@@ -306,13 +306,13 @@ def _env_update(makelinks, target_root, prev_mtimes, contents, env,
 		if not libdir_contents_changed:
 			makelinks = False
 
-	ldconfig = "/sbin/ldconfig"
+	ldconfig = eprefix + "/sbin/ldconfig"
 	if "CHOST" in settings and "CBUILD" in settings and \
 		settings["CHOST"] != settings["CBUILD"]:
 		ldconfig = find_binary("%s-ldconfig" % settings["CHOST"])
 
 	# Only run ldconfig as needed
-	if makelinks and ldconfig and not eprefix:
+	if makelinks and ldconfig:
 		# ldconfig has very different behaviour between FreeBSD and Linux
 		if ostype == "Linux" or ostype.lower().endswith("gnu"):
 			# We can't update links if we haven't cleaned other versions first, as
